source/transfer_list.rst

```diff
@@ -902,5 +903,134 @@ This manifest contains the SPMC attribute node consumed by the SPMD
      - hdr_size
      - Holds a SPMC manifest image in DT format.
 
+.. _dice_entries:
+
+Entries related to DICE
+^^^^^^^^^^^^^^^^^^^^^^^
+
+These entries hold [DICE]_ and [Open-DICE]_ information elements that needs to be transferred from one boot stage to the next in a *layered* DICE architecture, i.e., one that does not employ the DICE Protection Environment (DPE).
+
+This includes the Compound Device Identifier (CDI) used for attestation (and optional sealing), as well as the accumulated certificate chain.
+
+.. note::
+  As described in [DICE]_, CDIs are very sensitive data and are as long-lived as the layers they represent.
+  Each layer is therefore responsible for the proper handling of the CDIs it owns, as well as the derived cryptographic key material.
+  In particular, any given layer is expected to use and erase the CDI it received.
+  While the CDI is used, the layer needs the ability to prevent access to the CDI and disclosure of the value.
+
+The following entry types are currently defined:
+
+**CDI for attestation (XFERLIST_DICE_CDI_ATTEST)**
+
+This entry type holds the CDI described in [DICE]_.
+In [Open-DICE]_ it is referred to as "Attestation CDI".
+
+The CDI computed by the Sender's boot stage SHALL be added to the TL.
+The CDI received from the previous boot stage (i.e., DICE layer) MUST NOT be forwarded.
+
+.. _tab_dice_cdi_attest:
+.. list-table:: DICE attestation CDI
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x200**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size (in bytes) of the CDI
+
+   * - cdi_attest
+     - data_size
+     - hdr_size
+     - Holds the CDI used for attestation
+
+**CDI for sealing (XFERLIST_DICE_CDI_SEAL)**
+
+This entry type holds the "Sealing CDI" described in [Open-DICE]_.
+
+The CDI computed by the Sender's boot stage SHALL be added to the TL.
+The CDI received from the previous boot stage (i.e., DICE layer) MUST NOT be forwarded.
+
+.. _tab_dice_cdi_seal:
+.. list-table:: DICE sealing CDI
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x201**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size (in bytes) of the sealing CDI
+
+   * - cdi_seal
+     - data_size
+     - hdr_size
+     - Holds the CDI used for sealing
+
+**Certificate chain (XFERLIST_DICE_CERT_CHAIN)**
+
+This entry type holds the DICE certificate chain.
```

Comment:
What precisely is this? I guess a set of concatenated certificates with no metadata or padding between them? I wonder if it would be better for this to be a single certificate, with multiple instances representing the entire chain? That would make it easier for the chain to be extended as it's passed between components, or if only part of the chain needs to be passed around.
Also, what kind of certificates are represented here? The OpenDICE spec refers to UDS and CDI certificates so I guess you mean those? I presume there are also attestation and sealing certificates but the spec doesn't seem to specify those.

Reply:
> What precisely is this? I guess a set of concatenated certificates with no metadata or padding between them?

Yes, that was the idea.

> I wonder if it would be better for this to be a single certificate, with multiple instances representing the entire chain?

Good point.

> That would make it easier for the chain to be extended as it's passed between components, or if only part of the chain needs to be passed around.

I am not sure I can see a pruning use case, but your simplification argument is a good one.

> Also, what kind of certificates are represented here? The OpenDICE spec refers to UDS and CDI certificates so I guess you mean those?

Yes.

> I presume there are also attestation and sealing certificates but the spec doesn't seem to specify those.

The leaf cert in the chain is the "attestation" cert, i.e., the one that is supposed to sign challenges from the user.  The UDS cert is the "Device ID", while the rest are just intermediate CA certs.

There are no sealing certs, only sealing CDIs that can be used to derive sealing keys.

Reply:
OK, so can we expect another change to make this an OpenDICE certificate TE (instead of a certificate chain)?

> The leaf cert in the chain is the "attestation" cert

OK, so all certs are either OpenDICE UDS or CDI certs. Would be good to clarify this in the text.